client/api_challenge.go

```diff
@@ -18,7 +19,7 @@ import (
 )
 
 type Challenge interface {
-	GetChallengeInfo(ctx context.Context, info types.ChallengeInfo) (types.ChallengeResult, error)
+	GetChallengeInfo(ctx context.Context, info types.ChallengeInfo, opts types.GetChallengeInfoOptions) (types.ChallengeResult, error)
```

Comment:
I would suggests that the interface be changed to as below:
```
GetChallengeInfo(ctx context.Context, objectID string, pieceIndex, redundancyIndex int, opts types.GetChallengeInfoOptions)
```

Reply:
update as your suggestion